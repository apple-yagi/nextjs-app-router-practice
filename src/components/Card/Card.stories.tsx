import { Meta, StoryObj } from "@storybook/react";

import { Card } from ".";

const meta = {
  component: Card,
} satisfies Meta<typeof Card>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default = {
  args: {
    title: "新しく自分のブログサイトを作り直しました",
    ogImage:
      "https://images.microcms-assets.io/assets/a7208e73031e4a0c89212443979ab98a/82716655718e4eef8af491d11a52c66b/renewal_my_blog_ogp.png",
  },
} satisfies Story;
