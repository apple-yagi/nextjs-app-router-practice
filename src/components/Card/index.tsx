type Props = {
  title: string;
  ogImage: string;
};

export const Card = ({ title, ogImage }: Props) => {
  return (
    <article className="flex w-full max-w-xs flex-col items-center justify-center rounded-lg bg-white p-4 shadow-lg">
      <img width={288} height={140} src={ogImage} alt="" decoding="async" />
      <p className="mt-4 text-sm sm:text-base">{title}</p>
    </article>
  );
};
