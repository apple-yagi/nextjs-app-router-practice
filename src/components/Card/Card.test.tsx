import { composeStories } from "@storybook/react";
import { render } from "@testing-library/react";
import { describe, it } from "vitest";

import * as stories from "./Card.stories";

const { Default } = composeStories(stories);

describe("Card", () => {
  it("正しく表示されること", () => {
    expect(render(<Default />).asFragment()).toMatchSnapshot();
  });
});
