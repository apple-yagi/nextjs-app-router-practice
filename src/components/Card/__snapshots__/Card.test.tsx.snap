// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > 正しく表示されること 1`] = `
<DocumentFragment>
  <article
    class="flex w-full max-w-xs flex-col items-center justify-center rounded-lg bg-white p-4 shadow-lg"
  >
    <img
      alt=""
      decoding="async"
      height="140"
      src="https://images.microcms-assets.io/assets/a7208e73031e4a0c89212443979ab98a/82716655718e4eef8af491d11a52c66b/renewal_my_blog_ogp.png"
      width="288"
    />
    <p
      class="mt-4 text-sm sm:text-base"
    >
      新しく自分のブログサイトを作り直しました
    </p>
  </article>
</DocumentFragment>
`;
